src/web/resources/report.js

```diff
@@ -35,6 +35,111 @@ function Element(tagname, props, children) {
     return $elt;
 }
 
+const Filters = new Component("#filters", {
+    setup: function () {
+        // Warning fragile starting state
+        const regressedTags = ["uni-start", "lt-target", "lt-start", 
+        "apx-start", "timeout", "crash", "error"]
+        const improvedTags = ["imp-start", "ex-start", "eq-start", "eq-target",
+        "gt-target"]
+        const advancedTags = []
+        advancedTags.push(regressedTags)
+        advancedTags.push(improvedTags)
+
+        // build labels
+        const regressedLabel = this.buildCheckboxLabel("regressed","Regressed",true)
+        const improvedLabel = this.buildCheckboxLabel("improved","Improved",true)
+
+        // add listeners
+        regressedLabel.addEventListener("click", this.attachLeaderToChildren("regressed",regressedTags))
+        improvedLabel.addEventListener("click", this.attachLeaderToChildren("improved",improvedTags))
+
+        const advancedDiv = [this.buildChildren(improvedTags),this.buildChildren(regressedTags)]
+
+        this.elt.appendChild(Element("div", [new Text("Filters"),]))
+        this.elt.appendChild(Element("div", [improvedLabel, regressedLabel]))
+        this.elt.appendChild(Element("details",[Element("summary","Advanced"),advancedDiv]))
+    },
+    buildCheckboxLabel: function(idTag, text, boolState) {
+        return Element("label", {id:idTag},
+            [Element("input",
```

Comment:
I always put the `[` on the previous line, and I'd put the whole `input` element on one line.